src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_params.py

```diff
@@ -13,27 +13,27 @@
                                                 enum_choice_list)
 from .custom import (get_policy_completion_list, get_policy_assignment_completion_list,
                      get_resource_types_completion_list, get_providers_completion_list)
-from ._validators import validate_resource_type, validate_parent, resolve_resource_parameters, validate_deployment_name
+from ._validators import validate_deployment_name
 
 # BASIC PARAMETER CONFIGURATION
 
 resource_name_type = CliArgumentType(options_list=('--name', '-n'), help='The resource name.')
+_PROVIDER_HELP_TEXT = 'the resource namespace, aka \'provider\''
 
 register_cli_argument('resource', 'resource_name', resource_name_type)
 register_cli_argument('resource', 'api_version', help='The api version of the resource (omit for latest)', required=False)
-
-for item in ['tag', 'update', 'show', 'delete', 'exists']:
-    register_cli_argument('resource {}'.format(item), 'resource_provider_namespace', ignore_type)
-
-register_cli_argument('resource', 'resource_type', help='The resource type in <namespace>/<type> format.', type=validate_resource_type, validator=resolve_resource_parameters)
-register_cli_argument('resource', 'parent_resource_path', help='The parent resource type in <type>/<name> format.', type=validate_parent, required=False, options_list=('--parent',))
+register_cli_argument('resource', 'resource_id', options_list=('--id',), help='resource id')
+register_cli_argument('resource', 'resource_provider_namespace', options_list=('--namespace',), completer=get_providers_completion_list,
+                      help="resource namespaces like 'Microsoft.Network'. You can also specify it through --resource-type parameter")
+register_cli_argument('resource', 'resource_type', help="types such as 'virtualNetworks', or optional include namespace such as 'Microsoft.Network/virtualNetworks'")
+register_cli_argument('resource', 'parent_resource_path', required=False, options_list=('--parent',),
+                      help="path in formats of '<parent-type>/<parent-name>' or '<grandparent-type>/<grandparent-name>/<parent-type>/<parent-name>'")
```

Comment:
Here it might be better to give examples such as a SQL server and database path for the grandparent case and pretty much any normal one for the parent-type/parent-name case.